packages/python-packages/protocol-stub-generator/protocol/protocol_models.py

```diff
@@ -671,6 +794,398 @@ def to_json(self):
+                    m_type, key = get_map_type(yaml, name)
+                    m_type = format_type(m_type)
+                    if not key:
+                        key = self.parameters[0].name
+                    if inner_model:
+                        if index != -1:
+                            inner_model.append(
+                                Token(
+                                    key + "? : Map<string, " + param[0] + ">;",
+                                    TokenKind.Comment,
+                                )
+                            )
+                        else:
+                            inner_model.append(
+                                Token(
+                                    key + ": Map<string, " + param[0] + ">;",
+                                    TokenKind.Comment,
+                                )
+                            )
+                    else:
+                        if index != -1:
+                            self.add_comment(
+                                None, key + "? : Map<string, " + param[0] + ">;", None
+                            )
+                        else:
+                            self.add_comment(
+                                None, key + ": Map<string, " + param[0] + ">;", None
+                            )
+
+                        if in_model:
+                            self.add_new_line()
+                            self.add_whitespace(4)
+                            self.add_comment(None, "};", None)
+                            in_model = False
+                            indent = 4
+                else:
+                    if len(param) >= 2:
+                        if param[0] == "str":
+                            param[0] = "string"
+                        if param[0] == "any-object":
+                            param[0] = "{" + "}"
+                        if index != -1:
+                            json_request[i] = i + "? :" + param[0] + ";"
+                        else:
+                            json_request[i] = i + ": " + param[0] + ";"
+                        if inner_model:
+                            inner_model.append(
+                                Token(json_request[i], TokenKind.Comment)
+                            )
+                        else:
+                            self.add_comment(None, json_request[i], None)
+                            if in_model:
+                                self.add_new_line()
+                                self.add_whitespace(4)
+                                self.add_comment(None, "};", None)
+                                in_model = False
+                                indent = 4
+                    else:
+                        if json_request[i] == "str":
+                            json_request[i] = "string"
+                        if json_request[i] == "any-object":
+                            json_request[i] = "{" + "}"
```

Comment:
I've seen these same lines of code a few times. Convinced they should be refactored out :)